src/entities/score/services/__tests__/getCommitSummary.test.js

```diff
@@ -0,0 +1,68 @@
+import { describe, expect, it } from "vitest";
+import { getCommitSummary } from "..";
+
+describe("getCommitSummary", () => {
+  it("커밋 목록들을 받아서, 전체 점수와 100점인 커밋갯수를 올바르게 계산할 수 있어야한다.", () => {
+    const commitList = [
+      { qualityScore: 100 },
+      { qualityScore: 90 },
+      { qualityScore: 100 },
+    ];
+
+    const result = getCommitSummary(commitList);
+
+    expect(result).toEqual({
+      numOfPerfectCommits: 2,
+      totalScore: 96, // (100 + 90 + 100) / 3 = 96.666 => 96
+    });
+  });
+
+  it("검사한 모든 커밋목록들이 100점이라면, 총 점수는 100점이어야 한다.", () => {
+    const commitList = [
+      { qualityScore: 100 },
+      { qualityScore: 100 },
+      { qualityScore: 100 },
+    ];
+
+    const result = getCommitSummary(commitList);
+
+    expect(result).toEqual({
+      numOfPerfectCommits: 3,
+      totalScore: 100,
+    });
+  });
+
+  it("커밋목록들 중에, 100점인 커밋이 하나도 없어도 점수를 낼 수 있어야 한다.", () => {
+    const commitList = [
+      { qualityScore: 50 },
+      { qualityScore: 75 },
+      { qualityScore: 25 },
+    ];
+
+    const result = getCommitSummary(commitList);
+
+    expect(result).toEqual({
+      numOfPerfectCommits: 0,
+      totalScore: 50,
+    });
+  });
+
+  it("하나만 100점이었고, 총점이 소수점 내림으로 인해 0점이 된 경우에는 총점 1점으로 간주한다.", () => {
+    const generateZeroScore = (number) => {
+      const list = [];
+      for (let i = 0; i < number; i++) {
+        list.push({ qualityScore: 0 });
+      }
+      return list;
+    };
+
+    const commitList = [{ qualityScore: 100 }, ...generateZeroScore(100)];
```

Comment:
어레이에 한 요소만 추가하고, 순서가 중요하지는 않으므로 무거운 spread syntax 대신에 가벼운 push만 하셔도 될 것 같습니다!
```suggestion
    const commitList = generateZeroScore(100).push({ qualityScore: 100 });
```

Reply:
commitList에 push의 반환값이 들어가니 위 방식에서 `generateZeroScore()`  로 먼저 선언해주고 `{ qualityScore: 100 }` 을 push해주면 될 것 같습니다~!

Reply:
push 를 하면, 리턴값은 배열의 갯수가 되므로,

```suggestion
    const commitList = generateZeroScore(100);
    commitList.push({ qualityScore: 100 });
    const result = getCommitSummary(commitList);
```

이렇게 수정했습니다!

Reply:
수한님, 유진님 말씀이 맞네요! concat이랑 반환값을 헷갈렸습니다